include/revng/EarlyFunctionAnalysis/FunctionMetadataCache.h

```diff
@@ -48,45 +48,52 @@ extractFunctionMetadata(const llvm::BasicBlock *BB) {
 
 } // namespace detail
 
-class FunctionMetadataCache {
-private:
-  std::map<const llvm::Value *, efa::FunctionMetadata> FunctionCache;
+template<typename Traits>
```

Comment:
Can we have a concept for the trait (and `static_assert` it)?

Reply:
I added the concept but used it as a constraint instead of using `static_assert`. Let me know if you still prefer the `static_assert`.

Reply:
Ah sure. What I meant was:

```c++
// someheader.h
concept Lol = ...;

// myclass.h
// This class implements Lol
class MyClass { ... };
static_assert(Lol<MyClass>);
```

Reply:
So to be clear, you're okay with the constraint?